diff --git a/src/EventSubscriber/SetInlineBlockDependency.php b/src/EventSubscriber/SetInlineBlockDependency.php
new file mode 100644
index 0000000000000000000000000000000000000000..fcfb33789068fed3663fe82a2c7e7af3093df0cd
--- /dev/null
+++ b/src/EventSubscriber/SetInlineBlockDependency.php
@@ -0,0 +1,74 @@
+<?php
+
+namespace Drupal\layout_builder_st\EventSubscriber;
+
+use Drupal\block_content\BlockContentInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\layout_builder\EventSubscriber\SetInlineBlockDependency as SetInlineBlockDependencyBase;
+
+/**
+ * An event subscriber that returns an access dependency for inline blocks.
+ *
+ * This overrides the core version to also check revision IDs in translated
+ * components.
+ */
+class SetInlineBlockDependency extends SetInlineBlockDependencyBase {
+
+  /**
+   * {@inheritDoc}
+   *
+   * We override this to load the translated entity version if it is available.
+   */
+  protected function getInlineBlockDependency(BlockContentInterface $block_content, $operation): ?\Drupal\Core\Access\AccessibleInterface {
+    $layout_entity_info = $this->usage->getUsage($block_content->id());
+    if (empty($layout_entity_info)) {
+      // If the block does not have usage information then we cannot set a
+      // dependency. It may be used by another module besides layout builder.
+      return NULL;
+    }
+    $layout_entity = $this->entityRepository->getActive($layout_entity_info->layout_entity_type, $layout_entity_info->layout_entity_id);
+    // Overridden section starts here - we load the translation for the current
+    // language if it is available.
+    $current_langcode = \Drupal::languageManager()->getCurrentLanguage()->getId();
+    if ($layout_entity->hasTranslation($current_langcode)) {
+      $layout_entity = $layout_entity->getTranslation($current_langcode);
+    }
+    // Overridden section ends here.
+    if ($this->isLayoutCompatibleEntity($layout_entity)) {
+      if ($this->isBlockRevisionUsedInEntity($layout_entity, $block_content)) {
+        return $layout_entity;
+      }
+
+    }
+    return $layout_entity;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * We override this to check for revision IDs in translated components as
+   * well.
+   */
+  protected function isBlockRevisionUsedInEntity(EntityInterface $layout_entity, BlockContentInterface $block_content) {
+      $sections_blocks_revision_ids = [];
+      $translations = $layout_entity->getTranslationLanguages();
+      foreach ($translations as $langcode => $language) {
+          if ($layout_entity->hasTranslation($langcode)) {
+              $translated_entity = $layout_entity->getTranslation($langcode);
+              /** @var \Drupal\layout_builder_st\Plugin\SectionStorage\OverridesSectionStorage $section_storage */
+              $section_storage = $this->getSectionStorageForEntity($translated_entity);
+              $translated_sections = $this->getEntitySections($translated_entity);
+              $sections_blocks_revision_ids = array_merge($sections_blocks_revision_ids, $this->getInlineBlockRevisionIdsInSections($translated_sections));
+              foreach ($this->getInlineBlockComponents($translated_sections) as $component) {
+                  $configuration = $section_storage->getTranslatedComponentConfiguration($component->getUuid());
+                  if (!empty($configuration['block_revision_id'])) {
+                      $sections_blocks_revision_ids[] = $configuration['block_revision_id'];
+                  }
+              }
+              $sections_blocks_revision_ids = array_unique($sections_blocks_revision_ids);
+          }
+      }
+    return in_array($block_content->getRevisionId(), $sections_blocks_revision_ids);
+  }
+
+}
diff --git a/src/LayoutBuilderStServiceProvider.php b/src/LayoutBuilderStServiceProvider.php
index d079ce64da0021fd855dc2d9bd2baa6bcf8f1517..7943d9294b4e3881188f55ea7f80e7d9d46dda5e 100644
--- a/src/LayoutBuilderStServiceProvider.php
+++ b/src/LayoutBuilderStServiceProvider.php
@@ -4,6 +4,9 @@ namespace Drupal\layout_builder_st;
 
 use Drupal\Core\DependencyInjection\ContainerBuilder;
 use Drupal\Core\DependencyInjection\ServiceProviderBase;
+use Drupal\layout_builder_st\EventSubscriber\SetInlineBlockDependency;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
 
 class LayoutBuilderStServiceProvider extends ServiceProviderBase {
 
@@ -12,6 +15,20 @@ class LayoutBuilderStServiceProvider extends ServiceProviderBase {
    */
   public function register(ContainerBuilder $container) {
     $modules = $container->getParameter('container.modules');
+    if (isset($modules['block_content'])) {
+      $definition = new Definition(SetInlineBlockDependency::class);
+      //      EntityRepositoryInterface $entity_repository, Connection $database, InlineBlockUsageInterface $usage, SectionStorageManagerInterface $section_storage_manager, CurrentRouteMatch $current_route_match
+      $definition->setArguments([
+        new Reference('entity.repository'),
+        new Reference('database'),
+        new Reference('inline_block.usage'),
+        new Reference('plugin.manager.layout_builder.section_storage'),
+        new Reference('current_route_match')
+      ]);
+      $definition->addTag('event_subscriber');
+      $definition->setPublic(TRUE);
+      $container->setDefinition('layout_builder_st.get_block_dependency_subscriber', $definition);
+    }
 
     if (isset($modules['jsonapi'])) {
       $container
